controllers/feeds/getFeedsV2.js

```diff
@@ -191,13 +193,19 @@ module.exports = async (req, res) => {
           } else {
             item.is_self =
               item.actor.id === req.userId || item.actor.id === myAnonymousUser?.user_id;
-            let newItem = await modifyFeedIsFollowingTarget(item, req.userId);
-            newItem = await modifyAnonimityPost(newItem, isBlurredPost);
-            newItem = modifyReactionsPost(newItem, newItem.anonimity);
+            // check if actor is already in postActors
+            if (!postActors.includes(item.actor.id)) {
+              // post actor is a list of user id that already in the feed
+              // it will later be used to get their karma score
+              postActors.push(item.actor.id);
+            }
+
             if (item.verb === POST_VERB_POLL) {
               const postPoll = await modifyPollPostObject(req.userId, item, isBlurredPost);
               data.push(postPoll);
             } else {
+              let newItem = await isUserFollowAuthor(item, allFollowingUser);
+              newItem = modifyReactionsPost(newItem, newItem.anonimity);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

You're pushing actor IDs into `postActors` array without checking for duplicates. This could lead to unnecessary redundancy if an actor appears in multiple feeds. Consider using a Set instead of an Array for `postActors` to automatically handle duplicates.
```diff
- let postActors = [];
+ let postActors = new Set();

...

- if (!postActors.includes(item.actor.id)) {
-   postActors.push(item.actor.id);
- }
+ postActors.add(item.actor.id);

...

// Convert back to array when needed
- const karmaScores = await UsersFunction.getUsersKarmaScore(User, postActors);
+ const karmaScores = await UsersFunction.getUsersKarmaScore(User, Array.from(postActors));
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->